src/language_features/texlab.rs

```diff
@@ -0,0 +1,109 @@
+use crate::context::Context;
+use crate::position::get_lsp_position;
+use crate::types::{EditorMeta, EditorParams};
+use crate::PositionParams;
+use lsp_types::request::Request;
+use lsp_types::TextDocumentIdentifier;
+use lsp_types::TextDocumentPositionParams;
+use lsp_types::Url;
+use serde::{Deserialize, Serialize};
+use serde_repr::{Deserialize_repr, Serialize_repr};
+use std::fmt;
+
+pub enum ForwardSearch {}
+
+impl Request for ForwardSearch {
+    type Params = TextDocumentPositionParams;
+    type Result = ForwardSearchResult;
+    const METHOD: &'static str = "textDocument/forwardSearch";
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
```

Comment:
Nitpick :microscope: I think we usually don't use the camelCase annotation, unless there is a field that needs it.
So only for BuildTextDocumentParams, which has `text_document`, but not others.